connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginClassLoader.java

```diff
@@ -81,6 +85,48 @@ public String toString() {
         return "PluginClassLoader{pluginLocation=" + pluginLocation + "}";
     }
 
+    @Override
+    public URL getResource(String name) {
+        Objects.requireNonNull(name);
+
+        URL url = findResource(name);
+        if (url == null) {
+            url = super.getResource(name);
+        }
+        return url;
+    }
+
+    @Override
+    public Enumeration<URL> getResources(String name) throws IOException {
+        Objects.requireNonNull(name);
+        @SuppressWarnings("unchecked")
+        Enumeration<URL>[] tmp = (Enumeration<URL>[]) new Enumeration<?>[2];
+        tmp[0] = findResources(name);
+        // Explicitly call the parent implementation instead of super to avoid double-listing the local resources
+        tmp[1] = getParent().getResources(name);
+        return new Enumeration<URL>() {
+            private int index;
+            private boolean next() {
+                while (index < tmp.length) {
+                    if (tmp[index] != null && tmp[index].hasMoreElements()) {
+                        return true;
+                    }
+                    index++;
+                }
+                return false;
+            }
+            public boolean hasMoreElements() {
+                return next();
+            }
+            public URL nextElement() {
+                if (!next()) {
+                    throw new NoSuchElementException();
+                }
+                return tmp[index].nextElement();
+            }
+        };
```

Comment:
We can simplify this by using the `Vector` class, which is a `List` implementation capable of providing an `Enumeration` for its elements:
```suggestion
        Vector<URL> resources = new Vector<>();
        for (Enumeration<URL> foundLocally = findResources(name); foundLocally.hasMoreElements();) {
            URL url = foundLocally.nextElement();
            if (url != null)
                resources.add(url);
        }
        // Explicitly call the parent implementation instead of super to avoid double-listing the local resources
        for (Enumeration<URL> foundByParent = getParent().getResources(name); foundByParent.hasMoreElements();) {
            URL url = foundByParent.nextElement();
            if (url != null)
                resources.add(url);
        }
        return resources.elements();
```